### PR TITLE
add uat option to validator script and document in readme

### PR DESCRIPTION
**Pull request by niarenaw**
## Purpose
- Update dswx-s1-validator code to address R3 calval use cases
## Proposed Changes
- [ADD] endpoint option to specify OPS or UAT CMR venue
- [UPDATE] README to address new argument
- [UPDATE] README to fix iso timestamps in example

## Testing
- Code has been used with R3 calval data

### CODE REVIEW

**Comment by riverma** [comment]
Thanks a ton @niarenaw for your contribution! Let me test this out against some of our existing test cases and then get to approving / merging 👍

File: report/dswx-s1-validator/README.md [hunk 1]
```diff
@@ -58,15 +58,16 @@ This guide provides a quick way to get started with the script.
...
+6. Optionally, use the `--verbose` argument to get detailed information like a list of matching bursts and granule IDs
  
 ### Usage Examples
 
 * Retrieve a list of MGRS Tile Set IDs for the RTC burst processing a given time range on CMR, and filter the results to show only MGRS Tile Sets that had coverage of greater than or equal to 50%.
 
 ```
 $ python dswx_s1_validator.py --start "2023-12-05T01:00:00Z" --end "2023-12-05T03:59:59Z" --db MGRS_tile_collection_v0.2.sqlite --threshold 50
-Querying CMR for time range 2023-12-05T01:00:00Z to 2023-12-05T03:59:59Z.
+Querying CMR for time range 2023-12-05T01:00:00 to 2023-12-05T03:59:59.
```
Comment (riverma) [inline report/dswx-s1-validator/README.md:70]: I think we should keep the `Z` as that denotes times are expected to be in UTC.

File: report/dswx-s1-validator/dswx_s1_validator.py [hunk 2]
```diff
@@ -148,6 +148,7 @@ def get_total_granules(url, params, retries=5, backoff_factor=1):
     parser.add_argument("--file", required=False, help="Optional file path containing granule IDs")
     parser.add_argument("--threshold", required=False, help="Completion threshold minimum to filter results by (percentage format - leave out the %)")
     parser.add_argument("--verbose", action='store_true', help="Verbose and detailed output")
+    parser.add_argument("--endpoint", required=False, choices=['UAT', 'OPS'], default='OPS', help='CMR endpoint venue')
```
Comment (riverma) [inline report/dswx-s1-validator/dswx_s1_validator.py:151]: Nice that you've given a default 💯

File: report/dswx-s1-validator/dswx_s1_validator.py [hunk 3]
```diff
@@ -171,6 +172,8 @@ def get_total_granules(url, params, retries=5, backoff_factor=1):
 
         # Base URL for granule searches
         base_url = "https://cmr.earthdata.nasa.gov/search/granules.umm_json"
+        if args.endpoint == 'UAT':
```
Comment (riverma) [inline report/dswx-s1-validator/dswx_s1_validator.py:175]: 🥇

**Review by riverma — CHANGES_REQUESTED** [review]
Thank you for this contribution @niarenaw - excellent!

Couple suggestions inline but two I wanted to add:
- Did you make the regex fix to allow the `S1B` RTC granules to work? If so could you contribute that to this PR and rename the PR / purpose accordingly?
- UAT CMR has less granules than OPS, and I noticed that many times no granules are returned. I think we should add a `sys.exit(1)` right after [this line](https://github.com/nasa/opera-sds-pcm/blob/issue-703-r3-calval/report/dswx-s1-validator/dswx_s1_validator.py#L204) since now this is a common occurrence. Would you consider adding that?

File: report/dswx-s1-validator/README.md (hunk 1 above)
Comment (niarenaw) [reply report/dswx-s1-validator/README.md:70]: I get a ValueError, when including `Z`: `ValueError: Invalid isoformat string: '2023-12-05T01:00:00Z'`

**Comment by niarenaw** [comment]
@riverma thanks for you suggested changes -- I've added them.

**Review by hhlee445 — APPROVED** [review]

**Comment by hhlee445** [comment]
@niarenaw @riverma i'm trying to merge this PR, however, there are some conflict. @riverma can you help to resolve the conflicts?

**Comment by riverma** [comment]
Thanks @niarenaw for the additions! I've made a couple improvements to enhance the reliability. Was getting a number of granule query batch failures, but I think that's fixed now. Will merge. Thanks @hhlee445 as well.

**Review by riverma — APPROVED** [review]
Thanks @niarenaw for your help improving this script!

**Pull request merged**